pennylane/ops/qubit/hamiltonian.py

```diff
@@ -269,7 +269,7 @@ def terms(self):
         >>> t[0]
         [<tf.Tensor: shape=(), dtype=float32, numpy=1.0>, <tf.Tensor: shape=(), dtype=float32, numpy=2.0>]
         """
-        return self.coeffs, self.ops
+        return self.data, self.ops
```

Comment:
if we don't do this, we can't measure tensorflow hamiltonians (without patching it elsewhere). The difference:
```pycon
>>> coeffs = tf.Variable([0.1])
>>> ops = [qml.PauliZ(0)]
>>> H = qml.Hamiltonian(coeffs, ops)
>>> H.data
[<tf.Tensor: shape=(), dtype=float32, numpy=0.1>]
>>> H.coeffs
<tf.Variable 'Variable:0' shape=(1,) dtype=float32, numpy=array([0.1], dtype=float32)>
```

Reply:
Also, `convert_to_numpy_parameters` only updates the data, not the coeffs.